src/roiextractors/multiimagingextractor.py

```diff
@@ -0,0 +1,93 @@
+from abc import ABC
+from array import ArrayType
+from typing import Tuple
+
+import numpy as np
+
+from .extraction_tools import NumpyArray
+from .imagingextractor import ImagingExtractor
+
+
+class MultiImagingExtractor(ImagingExtractor, ABC):
+    """
+    This class is used to combine multiple ImagingExtractor objects by frame.
+    """
+
+    extractor_name = "MultiImagingExtractor"
+    installed = True
+    installation_mesg = ""
+
+    def __init__(self, imaging_extractors: list):
+        """
+        Parameters
+        ----------
+        imaging_extractors: list of ImagingExtractor
+            list of imaging extractor objects
+        """
+        super().__init__()
+        assert isinstance(imaging_extractors, list), "Enter a list of imaging extractor objects as argument"
+        assert all(isinstance(IX, ImagingExtractor) for IX in imaging_extractors)
+        self._imaging_extractors = imaging_extractors
+
+        # Num channels and sampling frequency based off the initial extractor
+        self._first_imaging_extractor = self._imaging_extractors[0]
+        self._num_channels = self._first_imaging_extractor.get_num_channels()
+        self._channel_names = self._first_imaging_extractor.get_channel_names()
+        self._sampling_frequency = self._first_imaging_extractor.get_sampling_frequency()
+        self._image_size = self._first_imaging_extractor.get_image_size()
+
```

Comment:
Thanks, good suggestion! Let me know what you think about the implementation
https://github.com/catalystneuro/roiextractors/blob/a400e0c404ae01868eb566453bed9bc56284cfed/src/roiextractors/multiimagingextractor.py#L51-L67

Reply:
```python
getattr(extractor, method)() if hasattr(extractor, method) else None
````

can be reduced to

```python
getattr(extractor, method, None)
```

Reply:
Also, to avoid confusion with metadata descriptions of properties may be better to rename the `property_desc` to something like `property_message`, since that's the role it plays in the format string below.

Reply:
Also, the `get_num_frames` need not be homogenous over the list of passed ImagingExtractors. Differing segments can be shorter or longer.